### PR TITLE
Add stricter error detection flags in bash scripts

### LINKED ISSUE #74: Eunomia-base image scripts don't handle errors well.
While running through the Hello Worl Yaml sample I get multiple errors. First, it appears the scripts use `find` and `cat`, neither of which exist in the image. Additionally, if the script attempts to delete resources that don't exist, either because its looking in the wrong namespace, or because the resources were already deleted, the script fails, leaving the CR stuck in delete mode as the finalizer was never cleaned up.

## What happens?

Run through the [Hello World YAML](https://github.com/KohlsTechnology/eunomia/tree/master/examples/hello-world-yaml) example. Then manually delete the resources created:

```
kubectl delete deployments.apps/hello-world
kubectl delete svc/hello-world
```

Now delete the cr, and look at the jobs logs from the resulting job.

```
$ kubectl logs gitopsconfig-hello-world-yaml-w4tydf-nfzj4
Cloning Repositories
Cloning into '/git/templates'...
Cloning into '/git/parameters'...
Setting cluster-ralated environment variable
Context "current" created.
Switched to context "current".
error: the server doesn't have a resource type "route"
Processing Templates
Managing Resources
Context "current" modified.
Switched to context "current".
cat: can't open 'find': No such file or directory
No resources found
cat: read error: Is a directory
No resources found
cat: unrecognized option: i
BusyBox v1.30.1 (2019-06-12 17:51:55 UTC) multi-call binary.

Usage: cat [-nbvteA] [FILE]...

Print FILEs to stdout

	-n	Number output lines
	-b	Number nonempty lines
	-v	Show nonprinting characters as ^x or M-x
	-t	...and tabs as ^I
	-e	...and end lines with $
	-A	Same as -vte
No resources found
cat: can't open '.*\.yaml': No such file or directory
No resources found
Error from server (NotFound): error when deleting "/git/manifests/hello-world.yaml": deployments.apps "hello-world" not found
Error from server (NotFound): error when deleting "/git/manifests/hello-world.yaml": services "hello-world" not found
```

## What were you expecting to happen?

...

## Steps to reproduce: 

* ...
* ...

## Any errors, stacktrace, logs? 

...

## Environment:

* Runtime version(Java, Go, Python, etc):
* Desktop OS/version:

## Additional comments:

...

### PR DESCRIPTION
**Pull request by akavel**
**Description**

This PR adds error detection flags (`set -euo pipefail`) in all bash scripts I was able to find in the repository, that were missing it before. This will improve robustness of the scripts.

Adding those flags resulted in a failing command being found in `discoverEnvironment.sh`. The PR adds a temporary workaround, and I opened a separate issue #152 with regards to fixing the underlying problem in the future.

Additionally, the PR enables execution tracing (`set -x`) in most of the scripts, including template-processors base scripts. This is intended to help when any debugging of the scripts is needed, and make it easier for contributors to provide better logs when reporting issues.

Closes #74 

**Type of change**

* Bug fix (non-breaking change which fixes an issue)

### CODE REVIEW

File: template-processors/base/bin/resourceManager.sh [hunk 1]
```diff
@@ -14,59 +14,54 @@
...
-    kube patch -R -f $MANIFEST_DIR
-  fi
-
+  case "$CREATE_MODE" in
+    CreateOrMerge)
+      kube apply -R -f "$MANIFEST_DIR"
+      ;;
+    CreateOrUpdate)
+      kube create -R -f "$MANIFEST_DIR"
+      kube update -R -f "$MANIFEST_DIR"
```
Comment (mkyc) [inline template-processors/base/bin/resourceManager.sh]: What about update operation? Did it clarify?

Comment (akavel) [reply template-processors/base/bin/resourceManager.sh]: I'm tracking this in a separate PR #149 so that any one of them does not block the other.

Comment (mkyc) [reply template-processors/base/bin/resourceManager.sh]: cool

**Comment by seanmalloy** [comment]
@akavel looks good. Please resolve the merge conflict, and we can get this merged.

**Comment by akavel** [comment]
@seanmalloy rebased & resolved, PTAL

**Pull request merged**